File: graphql-subscriptions-rails/app/graphql/types/post_type.rb
# frozen_string_literal: true

Types::PostType = GraphQL::ObjectType.define do
  name 'Post'
  description 'A blog post'

  field :id, !types.ID
  field :title, !types.String
  field :body, !types.String
  field :comments, types[!Types::CommentType]
end

File: graphql-subscriptions-express/graphql/post_type.js
const {GraphQLObjectType, GraphQLString, GraphQLInt} = require('graphql');
// const CommentType = require('./comment_type');

const PostType = new GraphQLObjectType({
    name: 'Post',
    fields: {
        id: {type: GraphQLInt},
        title: {type: GraphQLString},
        body: {type: GraphQLString}
    }
});

module.exports = PostType;
